src/hooks/useDownloadCsv.ts

```diff
@@ -0,0 +1,68 @@
+import { ApiClient, getEnvFromServer } from "../api/http-common";
+import buildSearchQuery, { TRouterQuery } from "@utils/buildSearchQuery";
+import { TLoadingStatus, TSearch } from "@types";
+import { useState } from "react";
+
+type TConfig = {
+  headers: {
+    accept: string;
+    "Content-Type": string;
+  };
+};
+
+const downloadFile = ({ data, fileName, fileType }) => {
+  const blob = new Blob([data], { type: fileType });
+
+  const a = document.createElement("a");
```

Comment:
Do we need to make sure this is hidden? Or if we do is the link not clickable? Have we tried this by enabling a mock of a slow connection in the browser debug? (to see how it behaves)